File: 0495.提莫攻击/0495-提莫攻击.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        return 0 if len(timeSeries) == 0 else sum([min(duration, timeSeries[i+1] - timeSeries[i])
                                                  for i in range(len(timeSeries)-1)] + [duration])
        """
        ret = 0
        for i in range(len(timeSeries)-1):
            ret += min(duration, timeSeries[i+1] - timeSeries[i])
        if len(timeSeries) > 0:
            ret += duration
        
        return ret
        """

File: 0495.提莫攻击/0495-提莫攻击.rs
impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
        if time_series.len() == 0 {
            return 0;
        }
        let mut ret = 0;
        let mut i: usize = 0;
        while i < time_series.len() - 1 {
            ret += (time_series[i+1] - time_series[i]).min( duration );
            i += 1;
        }
        if time_series.len() > 0 {
            ret += duration;
        }
        
        return ret;
    }
}